src/backend/columnar/cstore_metadata_tables.c

```diff
@@ -697,11 +698,14 @@ ReserveStripe(Relation rel, uint64 sizeBytes,
 
 	/*
 	 * We take ShareUpdateExclusiveLock here, so two space
-	 * reservations conflict, space reservation <-> vacuum
-	 * conflict, but space reservation doesn't conflict with
-	 * reads & writes.
+	 * reservations conflict. We use an advisory lock here
+	 * so we can easily detect these kind of locks in
+	 * IsProcessWaitingForSafeOperations() and don't include
+	 * them in the lock graph.
 	 */
-	LockRelation(rel, ShareUpdateExclusiveLock);
+	LOCKTAG tag;
+	SET_LOCKTAG_COLUMNAR_STRIPE_RESERVATION(tag, rel);
+	LockAcquire(&tag, ShareUpdateExclusiveLock, false, false);
```

Comment:
Nit: it would be helpful to make tiny helper functions to lock/unlock for columnar reservation. That would give a nice place to add comments.

Also, should we be using ExclusiveLock here? It seems like most callers do that, and that only relations are locked with SharedUpdateExclusive.

Reply:
done